observability/src/test/java/io/grpc/observability/interceptors/ConfigFilterHelperTest.java

```diff
@@ -0,0 +1,229 @@
+    assertFalse(configFilterHelper.globalLog);
+    assertFalse(configFilterHelper.methodOrServiceFilterPresent);
+    assertThat(configFilterHelper.perServiceFilters).isEmpty();
+    assertThat(configFilterHelper.perServiceFilters).isEmpty();
+    assertThat(configFilterHelper.perMethodFilters).isEmpty();
+    assertThat(configFilterHelper.logEventTypeSet).isEmpty();
+  }
+
+  @Test
+  public void enableCloudLogging_emptyLogFilters() {
+    configFilterHelper.setEmptyFilter();
+    when(mockConfig.isEnableCloudLogging()).thenReturn(true);
+    when(mockConfig.getLogFilters()).thenReturn(null);
+    when(mockConfig.getEventTypes()).thenReturn(null);
+    configFilterHelper.setMethodOrServiceFilterMaps();
+    configFilterHelper.setEventFilterSet();
+
+    assertFalse(configFilterHelper.globalLog);
+    assertFalse(configFilterHelper.methodOrServiceFilterPresent);
+    assertThat(configFilterHelper.perServiceFilters).isEmpty();
+    assertThat(configFilterHelper.perServiceFilters).isEmpty();
+    assertThat(configFilterHelper.perMethodFilters).isEmpty();
+    assertThat(configFilterHelper.logEventTypeSet).isEmpty();
+  }
+
+  @Test
+  public void enableCloudLogging_withLogFilters() {
+    configFilterHelper.setEmptyFilter();
+    when(mockConfig.isEnableCloudLogging()).thenReturn(true);
+    when(mockConfig.getLogFilters()).thenReturn(configLogFilters);
+    when(mockConfig.getEventTypes()).thenReturn(configEventTypes);
+
+    configFilterHelper.setMethodOrServiceFilterMaps();
+    configFilterHelper.setEventFilterSet();
+
+    assertTrue(configFilterHelper.globalLog);
+    MethodFilterParams expectedGlobalParams =
+        new MethodFilterParams(true, 128, 128);
+    assertThat(configFilterHelper.globalParams.log)
+        .isEqualTo(expectedGlobalParams.log);
+    assertThat(configFilterHelper.globalParams.headerBytes)
+        .isEqualTo(expectedGlobalParams.headerBytes);
+    assertThat(configFilterHelper.globalParams.messageBytes)
+        .isEqualTo(expectedGlobalParams.messageBytes);
+
+    assertTrue(configFilterHelper.methodOrServiceFilterPresent);
+    Map<String, MethodFilterParams> expectedServiceFilters = new HashMap<>();
+    expectedServiceFilters.put("service2",
+        new MethodFilterParams(true, 2048, 1024));
+
+    assertThat(configFilterHelper.perServiceFilters).hasSize(1);
+    assertThat(configFilterHelper.perServiceFilters.get("service2").log)
+        .isEqualTo(expectedServiceFilters.get("service2").log);
+    assertThat(configFilterHelper.perServiceFilters.get("service2").headerBytes)
+        .isEqualTo(expectedServiceFilters.get("service2").headerBytes);
+    assertThat(configFilterHelper.perServiceFilters.get("service2").messageBytes)
+        .isEqualTo(expectedServiceFilters.get("service2").messageBytes);
+
+    Map<String, MethodFilterParams> expectedMethodFilters = new HashMap<>();
+    expectedMethodFilters.put("service1/Method2",
+        new MethodFilterParams(true, 1024, 1024));
```

Comment:
Can you also add a test where you pass a method that's not included by the filters so you get `false` from the MethodFilterParams?

Reply:
Test `checkMethodNotToBeLogged()` tests when the method is not in the filter.